vertical-pod-autoscaler/enhancements/4016-in-place-updates-support/README.md

```diff
@@ -7,117 +7,175 @@
+* In the case of failure, make smaller updates to circumvent a node that does not have enough
+  headroom to accept the full resize but could accommodate a smaller one.
 
-### Applying Updates During Pod Admission
+We classify two types of updates in the context of this new mode:
 
-For VPAs in `InPlaceOnly` and `InPlaceOrRecreate` modes VPA Admission Controller will apply updates to starting pods,
-like it does for VPAs in `Initial`, `Auto`, and `Recreate` modes.
+1. Updates on pod admission
+2. In-place updates
 
-### Applying Disruption-free Updates
+### 1. Applying Updates During Pod Admission {#pod-admission}
 
-When an update only changes resources for which a container indicates that it doesn't require a restart, then VPA can
-attempt to actuate the change without disrupting the pod. VPA Updater will:
-* attempt to actuate such updates in place,
-* attempt to actuate them if difference between recommendation and request is at least 10%
-  * even if pod has been running for less than 12h,
-* if necessary execute only partial updates (if some changes would require a restart).
+For VPAs using the new `InPlaceOrRecreate` mode, the VPA Admission Controller will apply updates to
+starting pods just as it does for VPAs in `Initial`, `Auto`, and `Recreate` modes.
 
-### Applying Disruptive Updates
+### 2. In-Place Updates (**NEW**) {#in-place}
 
-In both `InPlaceOnly` and `InPlaceOrRecreate` modes VPA updater will attempt to apply updates that require container
-restart in place. It will update them under conditions that would trigger update with `Recreate` mode. That is it will
-apply disruptive updates if:
+In the `InPlaceOrRecreate` modes, and for updates that require a container restart, the VPA updater
+will attempt to apply updates in place. It will update them under the same conditions that would
+trigger an update with `Recreate` mode. That is it will apply an in-place update if:
 
 * Any container has a request below the corresponding `LowerBound` or
 * Any container has a request above the corresponding `UpperBound` or
 * Difference between sum of pods requests and sum of recommendation `Target`s is more than 10% and the pod has been
   running undisrupted for at least 12h.
-  * Successful disruptive update counts as disruption here (and prevents further disruptive updates to the pod for 12h).
+  * NOTE: A successful update counts as disruption here (and prevents further disruptive updates to the pod for 12h).
 
-In `InPlaceOrRecreate` mode (but not in `InPlaceOnly` mode) VPA updater will evict pod to actuate a recommendation if it
-attempted to apply the recommendation in place and failed.
+(NEW!) In addition, VPA will attempt an in-place update in some cases where we NORMALLY would not
+be able to perform an eviction, including:
+
+* If `CanEvict` is false.
+* If any of the `EvictionRequirements` on the VPA are not true.
+
+These additional resizes can be attempted because the eviction fallback would fail anyway.
```

Comment:
Just want to clarify this...
So an in-place update will be attempted, if that in-place fails, the VPA will not attempt an eviction? 
Almost as if it was `InPlaceOrNotAtAll` ?

Reply:
It will attempt an eviction but the eviction won't go through because the actual condition will prevent it from happening. Let me clarify that in the text.